File: vue/src/api/statusgroup.js
import request from '@/utils/request'

export function statusGroupList() {
  return request({
    url: '/statusgroup/list',
    method: 'post'
  })
}

export function addStatusGroup(data) {
  return request({
    url: '/statusgroup/add',
    method: 'post',
    data: data
  })
}

export function editStatusGroup(data) {
  return request({
    url: '/statusgroup/edit',
    method: 'post',
    data: data
  })
}

export function removeStatusGroup(id) {
  return request({
    url: '/statusgroup/remove',
    method: 'get',
    params: { id }
  })
}

export function getStatusGroupName() {
  return request({
    url: '/statusgroup/keyname',
    method: 'post'
  })
}

File: go/routegroup/setting/statusgroup.go
package setting

import (
	"itflow/handle"
	"itflow/midware"
	"itflow/model"

	"github.com/hyahm/xmux"
)

// StatusGroup 状态组路由
var StatusGroup *xmux.GroupRoute

func init() {
	StatusGroup = xmux.NewGroupRoute()

	StatusGroup.Post("/statusgroup/add", handle.AddStatusGroup).BindJson(&model.StatusGroup{})

	StatusGroup.Post("/statusgroup/edit", handle.EditStatusGroup).Bind(&model.StatusGroup{}).
		AddModule(midware.JsonToStruct)
	StatusGroup.Post("/statusgroup/list", handle.StatusGroupList)

	StatusGroup.Get("/statusgroup/remove", handle.DeleteStatusGroup)
	StatusGroup.Post("/statusgroup/keyname", handle.GetStatusGroupName)
}
